lib/bulk-apply-applications.js

```diff
@@ -243,91 +252,125 @@ function bulkApplyApplications (args, callback) {
+      } else {
+        eventDate = moment.utc(firstDate.startTime).format(eventDateFormat) + ' ' + startTime + ' - ' + endTime;
+      }
+      //common payload
+      var commonPayload = {
+        subjectVariables: [event.name],
+        locality: locality,
+        replyTo: dojo.email,
+        content: {
+          event: event,
+          dojo: dojo,
+          applicationDate: moment.utc(applications[0].created).format(eventDateFormat),
+          sessionName: session.name,
+          tickets: tickets,
+          status: applications[0].status,
+          applicationId: applications[iterator] ? applications[iterator].id : null,
+          eventDate: eventDate,
+          cancelLinkBase: protocol + '://' + zenHostname + '/dashboard/cancel_session_invitation'
+        }
+      };
+      async.parallel([
+        //Send parent and child emails
+        function sendApplicantsEmail () {
+          process.nextTick(function () {
+            emailSubject = emailSubjectData[applications[0].status];
+            async.eachSeries(profiles, function (profile, cb) {
+              //email payload changes for applicants
+              var payloadChanges = {
+                to: profile.email || null,
+                code: emailCode,
+                subject: emailSubject,
+                from: dojo.name+' <'+dojo.email+'>',
+                content: {
+                  applicantName: profile.name
+                }
+              };
+              //add common payload data
+              var payload = _.merge(payloadChanges, commonPayload);
+              iterator++;
+              if (!profile.email) return emailParents(cb);
 
-            function emailParents (cb) {
-              if (_.isEmpty(profile.parents)) return cb();
-              var parentsEmailed = [];
-              async.eachSeries(profile.parents, function (parent, cb) {
-                if (!_.isObject(parent)) {
-                  seneca.act({role: 'cd-users', cmd: 'load', id: parent, user: requestingUser}, function (err, parentUser) {
-                    if (err) return cb(err);
-                    payload.to = parentUser.email;
-                    if (!_.contains(parentsEmailed, payload.to) && payload.to !== profile.email) {
-                      parentsEmailed.push(payload.to);
-                      seneca.act({role: 'cd-dojos', cmd: 'send_email', payload: payload}, cb);
-                    } else {
-                      return cb();
-                    }
-                  });
-                } else {
+              //send the email
+              seneca.act({role: 'kue-queue', cmd: 'enqueue', name: 'bulk-apply-applications-kue', msg: _.clone({role: 'cd-dojos', cmd: 'send_email', payload: payload}, function (err, res) {
+                  if (err) return cb(err);
+                  if (applications[0].status === 'approved' || applications[0].status === 'pending' || applications[0].status === 'cancelled' ) {
```

Comment:
Only 'received' but I don't see the need for the if statement. Do we not want to email parents if the status is 'received'? If so surely it could be if (applications[0].status !== 'received') ?